### PR TITLE
Plane: tailsitter update

### PR DESCRIPTION
**Pull request by IamPete1**

### CODE REVIEW

File: plane/source/docs/guide-tailsitter.rst [hunk 1]
```diff
@@ -227,6 +151,67 @@ and thrust vectoring in each flight mode.
 .. youtube:: s2KLOAdS_HY
     :width: 100%
 
+PID gain scheduling
+===================
+
+There are a number of options for scaling control surface movement. Control surface effectiveness is dependant on airspeed they see, in tailsitter configurations this is heavily driven by the motor layout and prop wash.
```
Comment (nkruzan) [inline plane/source/docs/guide-tailsitter.rst:157]: dependent

File: plane/source/docs/guide-tailsitter.rst [hunk 2]
```diff
@@ -227,6 +151,67 @@ and thrust vectoring in each flight mode.
...
+should also be halved. Some tunning will be required for best performance, If oscillations are seen at high throttle :ref:`Q_TAILSIT_DSKLD<Q_TAILSIT_DSKLD>` should be reduced.
+
+For best results a airspeed sensor should be fitted.
+
+Note that Disk theory scheduling is only used if both Bit 0 and Bit 1 are disabled.
+
+Altitude correction (Bit 3)
+---------------------------
+
+Gain are scaled with altitude, this should be considered when operating over a wide range of altitudes, this method can be enabled in combination with any other method.
```
Comment (nkruzan) [inline plane/source/docs/guide-tailsitter.rst:192]: Gains are or Gain is, not sure which you intend

File: plane/source/docs/guide-tailsitter.rst [hunk 3]
```diff
@@ -227,6 +151,67 @@ and thrust vectoring in each flight mode.
...
+
+Disk theory (Bit 2)
+--------------------
+Disk theory gain scheduling attempts to calculate the airspeed seen on the control surfaces as a result of both prop wash and forward airspeed. In order for this calculation to be done :ref:`Q_TAILSIT_DSKLD<Q_TAILSIT_DSKLD>` must be set. This is the aircraft weight in KG divided by the total disk area of the propellers.
+The disk area for each propeller is calculated from the radius, the disk area of all propellers should then be summed. This allows ArduPilot to calculate the airspeed directly behind the propeller, however on a real vehicle 100% of the control surface is not in the direct prop wash. For example if half of the control surfaces are in the prop wash the calculated disk loading value 
+should also be halved. Some tunning will be required for best performance, If oscillations are seen at high throttle :ref:`Q_TAILSIT_DSKLD<Q_TAILSIT_DSKLD>` should be reduced.
+
+For best results a airspeed sensor should be fitted.
```
Comment (nkruzan) [inline plane/source/docs/guide-tailsitter.rst:185]: an airspeed

File: plane/source/docs/guide-tailsitter.rst [hunk 4]
```diff
@@ -284,3 +269,13 @@ dimension. When hovering it is important that there is not too much
 weight in the belly of the plane or on its back, so that it leans
 forward or back. This is particularly important for non-vectored
 tailsitters.
+
+Pre-Arm Issues
+==============
+
+Due to a issue in DCM related to compass fusion for yaw when pointing straight up, sometimes the AHRS subsystems will disagree when powering up, nose up. Slight errors in compass calibration, while resulting in a successful calibration, may worsen this effect.
```
Comment (nkruzan) [inline plane/source/docs/guide-tailsitter.rst:276]: an issue

File: plane/source/docs/guide-tailsitter.rst [hunk 5]
```diff
@@ -227,6 +151,67 @@ and thrust vectoring in each flight mode.
...
+
+Attitude/throttle based gain attenuation, with this option control surface deflection is reduced at high tilt angles and high throttle levels to prevent oscillation at high airspeeds. This can be used in combination with throttle scaling.
+
+Disk theory (Bit 2)
+--------------------
+Disk theory gain scheduling attempts to calculate the airspeed seen on the control surfaces as a result of both prop wash and forward airspeed. In order for this calculation to be done :ref:`Q_TAILSIT_DSKLD<Q_TAILSIT_DSKLD>` must be set. This is the aircraft weight in KG divided by the total disk area of the propellers.
+The disk area for each propeller is calculated from the radius, the disk area of all propellers should then be summed. This allows ArduPilot to calculate the airspeed directly behind the propeller, however on a real vehicle 100% of the control surface is not in the direct prop wash. For example if half of the control surfaces are in the prop wash the calculated disk loading value 
+should also be halved. Some tunning will be required for best performance, If oscillations are seen at high throttle :ref:`Q_TAILSIT_DSKLD<Q_TAILSIT_DSKLD>` should be reduced.
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:183]: s/tunning/tuning

**Review by kd0aij — COMMENTED** [review]
LGTM; just noticed one typo

**Comment by Hwurzburg** [comment]
@IamPete1 Pete, I have a few format and expansion things that I will do after I merge rather than modify this...but before I merge, do we need to still have the NY plus and X figures here since they have different motor assignments than the copter ones? I can re-add after I merge this and speak to those frame types

**Comment by IamPete1** [comment]
@Hwurzburg The no yaw torque frame are gone, I don't think it is even possible on 4.1 any more. Without a custom frame from scripting anyway....

**Comment by Hwurzburg** [comment]
> @Hwurzburg The no yaw torque frame are gone, I don't think it is even possible on 4.1 any more. Without a custom frame from scripting anyway....

why are they still listed in Q_FRAME_TYPE :

16 | MOTOR_FRAME_TYPE_NYT_PLUS
-- | --
17 | MOTOR_FRAME_TYPE_NYT_X

**Comment by kd0aij** [comment]
The NYT frame types should still work. The NYT_PLUS would be appropriate for counter-rotating motors on the wing.
You can fly the bi-wing either way, but I don't recall anyone investigating the pros and cons for that planform in a real aircraft.

**Comment by IamPete1** [comment]
@Hwurzburg @kd0aij My apology's your both correct, I thought they were lost when we went back to the 'normal' motors matrix for tailsitters.

**Comment by Hwurzburg** [comment]
@IamPete1 Added another commit expanding, minor style and format.....if changes are okay I will merge this

**Comment by IamPete1** [comment]
@Hwurzburg Looks good to me, thank you!

File: plane/source/docs/guide-tailsitter.rst [hunk 6]
```diff
@@ -71,43 +95,39 @@ Motors are controlled by the Throttle, Throttle Left, Throttle Right outputs:
...
 Key parameters
 ==============
 
-The key differences between fixed wing flight and hovering for a
+The key differences between fixed wing flight and VTOL for a
 tailsitter are:
 
-- when hovering the copter PID gains will be used (the ones starting
-  with Q_A_RAT_*)
+- when in VTOL the copter stle PID gains will be used (the ones starting
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:116]: style

File: plane/source/docs/guide-tailsitter.rst [hunk 7]
```diff
@@ -71,43 +95,39 @@ Motors are controlled by the Throttle, Throttle Left, Throttle Right outputs:
...
-This should be set if your tailsitter can achieve much larger pitch angle than 
-would be safe for roll (some airframes can't recover from high-speed knife-edge flight).
-
-:ref:`Q_TRIM_PITCH<Q_TRIM_PITCH>` can be used to account for any offset in hover pitch angle due to the center of gravity being closer to the belly or top of the fuselage. Set this such that the the vehicle does not drift forwards or backwards in QSTABILIZE or QHOVER with no pilot inputs.
+  ``RLL_RATE_x`` and ``PTCH_RATE_x`` gains)
+- :ref:`Q_TAILSIT_RLL_MX<Q_TAILSIT_RLL_MX>` allows the roll limit angle limit to be set differently from :ref:`Q_ANGLE_MAX<Q_ANGLE_MAX>`. If left at zero, both pitch and roll are limited by :ref:`Q_ANGLE_MAX<Q_ANGLE_MAX>`. If :ref:`Q_TAILSIT_RLL_MX<Q_TAILSIT_RLL_MX>` is nonzero roll angle will be limited and pitch max angle will still be :ref:`Q_ANGLE_MAX<Q_ANGLE_MAX>`. This should be set if your tailsitter can achieve much larger pitch angle than would be safe for roll (some airframes can't recover from high-speed knife-edge flight).
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:120]: (some airframes can't recover from high-speed knife-edge flight using only yaw control)

File: plane/source/docs/guide-tailsitter.rst [hunk 8]
```diff
@@ -71,43 +95,39 @@ Motors are controlled by the Throttle, Throttle Left, Throttle Right outputs:
...
+- :ref:`Q_TRIM_PITCH<Q_TRIM_PITCH>` can be used to account for any offset in hovering pitch angle due to the center of gravity being closer to the belly or top of the fuselage, or to counter fixed wing AHRS trim used to set angle of attack in cruise, instead of using :ref:`TRIM_PITCH_CD<TRIM_PITCH_CD>`. Set this such that the the vehicle does not drift forwards or backwards in QSTABILIZE or QHOVER with no pilot inputs.
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:121]: should this be "thrust line not through the CG"?

File: plane/source/docs/guide-tailsitter.rst [hunk 9]
```diff
@@ -151,40 +176,40 @@ and thrust vectoring in each flight mode.
 .. youtube:: s2KLOAdS_HY
     :width: 100%
 
-PID gain scheduling
-===================
+CopterMotor PID Gain Scaling
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:179]: a google search for "pid gain scheduling" produces more relevant results than one for "pid gain scaling"
But that's probably a dev-level concern :)

File: plane/source/docs/guide-tailsitter.rst [hunk 10]
```diff
@@ -243,9 +272,9 @@ interpreted using the :ref:`Q_TAILSIT_INPUT<Q_TAILSIT_INPUT>` parameter. The cho
 Tailsitter Input Mask
 =====================
 
-To support people flying 3D aircraft and wanting to learn how to
-prop-hang manually, you can set the :ref:`Q_TAILSIT_MASK<Q_TAILSIT_MASK>` to a mask of
-channels that will have full manual input control while hovering.
+To support people with experience flying 3D aircraft and wanting to learn how to
+prop-hang manually, you can set the :ref:`Q_TAILSIT_MASK<Q_TAILSIT_MASK>` to determine which
+channels that will have full manual input control without attitude stabilization while hovering in QHOVER and QSTABILIZE.
```
Comment (kd0aij) [inline plane/source/docs/guide-tailsitter.rst:277]: determine which channels that ~~will~~ have full

**Review by kd0aij — COMMENTED** [review]
some minor suggestions

File: plane/source/docs/guide-tailsitter.rst (hunk 10 above)
Comment (Hwurzburg) [reply plane/source/docs/guide-tailsitter.rst:277]: done

File: plane/source/docs/guide-tailsitter.rst (hunk 9 above)
Comment (Hwurzburg) [reply plane/source/docs/guide-tailsitter.rst:179]: scaling is used through-out the wiki and in hobbyist circles...made it consistent

File: plane/source/docs/guide-tailsitter.rst (hunk 8 above)
Comment (Hwurzburg) [reply plane/source/docs/guide-tailsitter.rst:121]: done

File: plane/source/docs/guide-tailsitter.rst (hunk 7 above)
Comment (Hwurzburg) [reply plane/source/docs/guide-tailsitter.rst:120]: done

File: plane/source/docs/guide-tailsitter.rst (hunk 6 above)
Comment (Hwurzburg) [reply plane/source/docs/guide-tailsitter.rst:116]: nice catch

**Comment by Hwurzburg** [comment]
> some minor suggestions

thanks!

**Comment by IamPete1** [comment]
@Hwurzburg @kd0aij Thank you both for your help getting this in.

**Pull request merged**